### ISSUE TITLE
cargo_common_metadata shouldn't warn on `publish = false`

### ISSUE THREAD

**Issue by daxpedda** [rust-lang/rust-clippy#6649]
**Lint name:** cargo_common_metadata

**I tried this code:**
```rust
[package]
name = "cargo_common_metadata"
version = "0.1.0"
publish = false
```

**I expected to see this happen:**
I expect no errors to happen, because `publish` is set to false.
Sometimes you have crates in your workspace that you only use for testing or other purposes, which don't need any of the metadata, this is why they have `publish = false`.

**Instead, this happened:**
<details><summary>Warnings:</summary>

```
error: package `cargo_common_metadata` is missing `package.authors` metadata
   |
   = note: `-D clippy::cargo-common-metadata` implied by `-D warnings`

error: package `cargo_common_metadata` is missing `package.description` metadata

error: package `cargo_common_metadata` is missing `either package.license or package.license_file` metadata

error: package `cargo_common_metadata` is missing `package.repository` metadata

error: package `cargo_common_metadata` is missing `package.readme` metadata

error: package `cargo_common_metadata` is missing `package.keywords` metadata

error: package `cargo_common_metadata` is missing `package.categories` metadata

error: aborting due to 7 previous errors
```
</details>

### Meta

- `cargo clippy -V`: e.g. clippy 0.0.212 (f455e46 2020-06-20)
- `rustc -Vv`:
  ```
  rustc 1.51.0-nightly (e38fb306b 2021-01-14)
  binary: rustc
  commit-hash: e38fb306b7f5e65cca34df2dab1f0db15e1defb4
  commit-date: 2021-01-14
  host: x86_64-unknown-linux-gnu
  release: 1.51.0-nightly
  LLVM version: 11.0
  ```
